src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java

```diff
@@ -412,6 +427,175 @@ public void deleteOicHearingResult(
         oicHearingResultRepository.delete(oicHearingResult);
     }
 
+    @Getter
+    @Setter
+    @AllArgsConstructor
+    class OicSanctionValidationResult {
+        Adjudication adjudication;
+        Long oicHearingId;
+        Long offenderBookId;
+    }
+
+    protected OicSanctionValidationResult validateOicSanction(Long adjudicationNumber) {
+        final var adjudication = adjudicationsRepository.findByParties_AdjudicationNumber(adjudicationNumber)
+            .orElseThrow(EntityNotFoundException.withMessage(format("Could not find adjudication number %d", adjudicationNumber)));
+
+        final var hearingResult = oicHearingResultRepository.findByAgencyIncidentIdAndFindingCode(adjudication.getAgencyIncidentId(), FindingCode.PROVED);
+        if (hearingResult.isEmpty()) throw EntityNotFoundException.withMessage(format("Could not find hearing result PROVED for adjudication id %d", adjudication.getAgencyIncidentId()));
+        if (hearingResult.size() > 1) throw EntityNotFoundException.withMessage(format("Multiple PROVED hearing results for adjudication id %d", adjudication.getAgencyIncidentId()));
+
+        return new OicSanctionValidationResult(
+            adjudication,
+            hearingResult.get(0).getOicHearingId(),
+            adjudication.getOffenderParty().get().getOffenderBooking().getBookingId()
+        );
+    }
+
+    @Transactional
+    @VerifyOffenderAccess
+    public List<Sanction> createOicSanctions(
+        final Long adjudicationNumber,
+        final List<OicSanctionRequest> oicSanctionRequests) {
+
+        var result = validateOicSanction(adjudicationNumber);
+
+        Long offenderBookId = result.adjudication.getOffenderParty().get().getOffenderBooking().getBookingId();
+        Long nextSanctionSeq = oicSanctionRepository.getNextSanctionSeq(offenderBookId);
+
+        List<OicSanction> oicSanctions = new ArrayList<>();
+        int index = 0;
+        for (var request : oicSanctionRequests) {
+            oicSanctions.add(oicSanctionRepository.save(OicSanction.builder()
```

Comment:
FYI an alternative approach would be to just model in JPA, sanctions as a 1 to many of a hearing result (?) and just add them all in Java to a `findBy` result  rather than using oicSanctionRepository.save - but both methods are completely valid